### PR TITLE
(Refactor) unit tests && refactoring

### PR DESCRIPTION
**Pull request by vbaranov**
unit tests && refactoring for `generateInitalKeys` script for clones of POA Network.
This approach will be implemented also for `poa-scripts-validators` repos and for all DApps of POA Network

- Unit tests are written with Mocha
- POA Network setup is created with https://github.com/poanetwork/poa-test-setup
- unit tests simply start with `npm test` command. POA Network Setup is configured and tests are executed on it.
- .travis.yml is added to this repo. Travis CI can be configured for this repo.
- refactoring affects the handling of errors, that was not handled before
- `node generetaInititalKey` -> `npm start`

The same PR will be created for Sokol branch after merging.

### CODE REVIEW

File: generateInitialKey/test/test.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+	it('initial key should be valid to generate production keys', async function() {
+		this.timeout(10000);
+		initialKey = `0x${keyStore.address}`
+		
+		var privateKey = keythereum.recover(password, keyStore);
+		const privateKeyHex = Buffer.from(privateKey, 'hex')
+
+		try { config = await getConfig() }
+		catch (err) { return errorFinish(err); }
```
Comment (phahulin) [inline generateInitialKey/test/test.js]: `errorFinish` is probably undefined in this file

File: generateInitialKey/test/test.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+		rpc = process.env.RPC || config.Ethereum[config.environment].rpc || 'http://127.0.0.1:8545'
+
+		try { web3 = await configureWeb3(rpc) }
+		catch (err) { return errorFinish(err); }
```
Comment (phahulin) [inline generateInitialKey/test/test.js:69]: also here `errorFinish` is undefined

File: generateInitialKey/test/test.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+		    for (let i in files) {
+		    	let file = files[i]
+		    	let ext = path.extname(file)
+		    	if (!listOfExt.includes(ext)) reject({message: `Incorrect ext ${ext} of file`})
+
+		    	if (ext == jsonExt) {
+		    		keyStorePath = file
+		    		try { keyStore = await checkKeyStoreFile(file) }
+					catch (err) { return reject(err) }
+		    	} else if (ext = keyExt) {
```
Comment (phahulin) [inline generateInitialKey/test/test.js]: should be `ext == keyExt` otherwise always true

File: generateInitialKey/test/test.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+			try {
+			  	resolve(content)
+			} catch (e) {
+			  	reject(e)
+			}
+		});
+	});
+}
+
+function sendRawTX() {
```
Comment (phahulin) [inline generateInitialKey/test/test.js]: this function is probably here by mistake - it is never used and contains undefined variables `serializedTx`, `response` and function `sendRawTransactionResponse`

File: generateInitialKey/test/test.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+	})
+
+	it('initial key should be valid to generate production keys', async function() {
+		this.timeout(10000);
+		initialKey = `0x${keyStore.address}`
+		
+		var privateKey = keythereum.recover(password, keyStore);
+		const privateKeyHex = Buffer.from(privateKey, 'hex')
+
+		try { config = await getConfig() }
```
Comment (phahulin) [inline generateInitialKey/test/test.js]: `config` is defined as a global variable here

File: generateInitialKey/test/test.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+		    	}
+		    }
+		    resolve();
+		});
+	});
+}
+
+function checkKeyStoreFile(file) {
+	return new Promise(async (resolve, reject) => {
+		try { content = await readFile(file) }
```
Comment (phahulin) [inline generateInitialKey/test/test.js]: `content` is defined as a global variable

File: generateInitialKey/utils/configureWeb3.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+const Web3 = require('web3');
+const { errorFinish } = require('./errorResponse');
+
+//configures web3
+function configureWeb3(rpc) {
+	let web3;
+	if (typeof web3 !== 'undefined') web3 = new Web3(web3.currentProvider);
+	else web3 = new Web3(new Web3.providers.HttpProvider(rpc));
+
+	if (!web3) return errorFinish(err);
```
Comment (phahulin) [inline generateInitialKey/utils/configureWeb3.js]: `err` is not defined here

File: generateInitialKey/utils/generateAddress.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+const keythereum = require("keythereum");
+
+//generates initial key keystore file
+function generateAddress(password) {
+	return new Promise((resolve, reject) => {
+		let params = { keyBytes: 32, ivBytes: 16 };
+
+	  	let dk = keythereum.create(params);
+
+	  	keythereum.create(params, function (dk) {
```
Comment (phahulin) [inline generateInitialKey/utils/generateAddress.js]: `dk` here shadows `dk` defined above (at line 8). Either this one should be renamed, or the one above is never used.

**Review by phahulin — COMMENTED** [review]
Errors returned by eslint

**Comment by vbaranov** [comment]
@phahulin fixed. Thanks!

**Review by phahulin — APPROVED** [review]
LGTM

**Pull request merged**